src/config.py

```diff
@@ -151,6 +152,38 @@ def auth_properties(self) -> List[str]:
             f'listener.name.sasl_plaintext.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
         ]
 
+    @property
+    def tls_properties(self) -> List[str]:
+        """Builds the properties necessary for TLS authentication.
+
+        Returns:
+            list of properties to be set
+        """
+        host = (
+            self.charm.model.get_relation(PEER).data[self.charm.unit].get("private-address", None)
+        )
+
+        f"zookeeper.connect={host}:"
+        return [
+            f'listener.name.sasl_ssl.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
+            "listeners=SASL_SSL://:9093",
+            f"advertised.listeners=SASL_SSL://{host}:9093",
+            "zookeeper.ssl.client.enable=true",
+            f"zookeeper.ssl.truststore.location={self.truststore_filepath}",
+            f"zookeeper.ssl.truststore.password={self.charm.tls.truststore_password}",
+            "zookeeper.clientCnxnSocket=org.apache.zookeeper.ClientCnxnSocketNetty",
+            "security.inter.broker.protocol=SASL_SSL",
+            "security.protocol=SASL_SSL",
+            f"ssl.truststore.location={self.truststore_filepath}",
+            f"ssl.truststore.password={self.charm.tls.truststore_password}",
+            f"ssl.keystore.location={self.keystore_filepath}",
+            f"ssl.keystore.password={self.charm.tls.keystore_password}",
+            f"ssl.key.password={self.charm.tls.keystore_password}",
+            "zookeeper.ssl.endpoint.identification.algorithm=",
+            "ssl.endpoint.identification.algorithm=",
+            "ssl.client.auth=none",
```

Comment:
Let's keep any non-inter-broker/ZK changes for now, but not implement client SSL yet, as there's probably some work needed for how we want to inform/update clients to the new port + remove PLAINTEXT port.

Perhaps better yet, the inter-broker + ZK SSL config options could be their own `@property`, and the client ones another. And then in this PR we could just omit the latter for the time being.

In this case, that would mean removing all the SSL specific:

```
listener.name.sasl_ssl...
listeners
advertised.listeners
security.protocol
ssl.keystore.location
ssl.keystore.password
ssl.endpoint.identification.algorithm
ssl.client.auth
```

Reply:
I also think we can remove `ssl.key.password` completely, as we're not setting a private-key password.